File: app/src/main/java/com/prey/activities/WelcomeActivity.java
/*******************************************************************************
 * Created by Carlos Yaconi
 * Copyright 2015 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.activities;

import android.content.Intent;
import android.content.pm.ActivityInfo;
import android.content.res.Configuration;
import android.os.Bundle;
import android.view.Window;

import androidx.fragment.app.FragmentActivity;

import com.prey.PreyConfig;
import com.prey.PreyLogger;
import com.prey.backwardcompatibility.FroyoSupport;

public class WelcomeActivity extends FragmentActivity {

    @Override
    public void onResume() {
        PreyLogger.d("onResume of WelcomeActivity");
        super.onResume();
        menu();
    }

    @Override
    public void onPause() {
        PreyLogger.d("onPause of WelcomeActivity");
        super.onPause();
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
        PreyLogger.d("onCreate of WelcomeActivity");
        menu();

    }


    public void menu() {
        PreyLogger.d("menu WelcomeActivity");
        String email = PreyConfig.getPreyConfig(this).getEmail();
        if(email==null||"".equals(email)){
            PreyLogger.d("email:"+email);
            PreyConfig.getPreyConfig(this).setProtectReady(false);
            PreyConfig.getPreyConfig(this).setProtectAccount(false);
            PreyConfig.getPreyConfig(this).setProtectTour(false);
        }
        if (PreyConfig.getPreyConfig(this).isThisDeviceAlreadyRegisteredWithPrey()){
            ready();
        } else {
            Intent intent = new Intent(getApplicationContext(), InitActivity.class);
            startActivity(intent);
            finish();
        }
    }

    public void ready() {
        PreyLogger.d("ready WelcomeActivity");
        Intent intent = new Intent(getApplicationContext(), CheckPasswordHtmlActivity.class);
        startActivity(intent);
        finish();
    }

    private static final int SECURITY_PRIVILEGES = 10;

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        PreyLogger.d("requestCode:" + requestCode + " resultCode:" + resultCode);
        if (requestCode == SECURITY_PRIVILEGES) {
            menu();
            PreyConfig.getPreyConfig(getApplicationContext()).setProtectPrivileges(true);
        }
    }

    public void addPrivileges() {
        Intent intent = FroyoSupport.getInstance(getApplicationContext()).getAskForAdminPrivilegesIntent();
        startActivityForResult(intent, SECURITY_PRIVILEGES);
    }





}

File: app/src/main/java/com/prey/activities/kotlin/WelcomeActivity.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.activities.kotlin

import android.content.Intent
import android.content.pm.ActivityInfo
import android.content.res.Configuration
import android.os.Build
import android.os.Bundle
import android.view.Window
import androidx.fragment.app.FragmentActivity
import com.prey.backwardcompatibility.kotlin.FroyoSupport
import com.prey.kotlin.PreyConfig
import com.prey.kotlin.PreyLogger

class WelcomeActivity : FragmentActivity() {
    public override fun onResume() {
        PreyLogger.d("onResume of WelcomeActivity")
        super.onResume()
        menu()
    }

    public override fun onPause() {
        PreyLogger.d("onPause of WelcomeActivity")
        super.onPause()
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        PreyLogger.d("onCreate of WelcomeActivity")
        menu()
    }

    fun menu() {
        PreyLogger.d("menu WelcomeActivity")
        val email: String? = PreyConfig.getInstance(this).getEmail()
        if (email == null || "" == email) {
            PreyLogger.d("email:$email")
            PreyConfig.getInstance(this).setProtectReady(false)
            PreyConfig.getInstance(this).setProtectAccount(false)
            PreyConfig.getInstance(this).setProtectTour(false)
        }
        if (PreyConfig.getInstance(this).isThisDeviceAlreadyRegisteredWithPrey()) {
            ready()
        } else {
            val intent = Intent(applicationContext, LoginActivity::class.java)
            startActivity(intent)
            finish()
        }
    }

    fun ready() {
        PreyLogger.d("ready WelcomeActivity")
        var intent: Intent? = null
        intent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            Intent(applicationContext, CheckPasswordHtmlActivity::class.java)
        } else {
            Intent(applicationContext, DeviceReadyActivity::class.java)
        }
        startActivity(intent)
        finish()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        PreyLogger.d("requestCode:$requestCode resultCode:$resultCode")
        if (requestCode == SECURITY_PRIVILEGES) {
            menu()
            PreyConfig.getInstance(applicationContext).setProtectPrivileges(true)
        }
    }

    fun addPrivileges() {
        val intent = FroyoSupport.getInstance(applicationContext)!!.askForAdminPrivilegesIntent
        startActivityForResult(intent, SECURITY_PRIVILEGES)
    }

    companion object {
        private const val SECURITY_PRIVILEGES = 10
    }
}